sympy/physics/optics/gaussopt.py

```diff
@@ -526,18 +527,19 @@ class BeamParameter(Expr):
     # subclass it. See:
     # https://groups.google.com/d/topic/sympy/7XkU07NRBEs/discussion
 
-    def __new__(cls, wavelen, z, z_r=None, w=None):
+    def __new__(cls, wavelen, z, n=1, z_r=None, w=None):
```

Comment:
`n=1` should be added at the end of the list so arguments passed by position still works. As it is now, this would be backwards incompatible.

Reply:
Yes I had tried it out for quite some time while making the initial commit but it failed `test_args.py` giving -
Traceback (most recent call last):
```
  File "c:\users\kunni\sympy\sympy\sympy\core\tests\test_args.py", line 4915, in test_sympy__physics__optics__gaussopt__BeamParameter
    assert _test_args(BeamParameter(530e-9, 1, w=1e-3, n=1))
  File "c:\users\kunni\sympy\sympy\sympy\core\tests\test_args.py", line 101, in _test_args
    recreatable = not obj.args or obj.func(*obj.args) == obj
  File "c:\users\kunni\sympy\sympy\sympy\physics\optics\gaussopt.py", line 540, in __new__
    raise ValueError('Constructor expects exactly one named argument.')
ValueError: Constructor expects exactly one named argument.
```

Is there any necessary change that I miss other than simply keeping n=1 at the end of the list uniformly throughout the code.
Thanks.

Reply:
```diff
diff --git a/sympy/physics/optics/gaussopt.py b/sympy/physics/optics/gaussopt.py
index f89e374312..56037ff816 100644
--- a/sympy/physics/optics/gaussopt.py
+++ b/sympy/physics/optics/gaussopt.py
@@ -527,7 +527,7 @@ class BeamParameter(Expr):
     # subclass it. See:
     # https://groups.google.com/d/topic/sympy/7XkU07NRBEs/discussion

-    def __new__(cls, wavelen, z, n=1, z_r=None, w=None):
+    def __new__(cls, wavelen, z, z_r=None, w=None, n=1):
         wavelen = sympify(wavelen)
         z = sympify(z)
         n = sympify(n)
@@ -536,10 +536,10 @@ def __new__(cls, wavelen, z, n=1, z_r=None, w=None):
             z_r = sympify(z_r)
         elif w is not None and z_r is None:
             z_r = waist2rayleigh(sympify(w), wavelen, n)
-        else:
-            raise ValueError('Constructor expects exactly one named argument.')
+        elif z_r is None and w is None:
+            raise ValueError('Must specify one of w and z_r.')

-        return Expr.__new__(cls, wavelen, z, n, z_r)
+        return Expr.__new__(cls, wavelen, z, z_r, n)

     @property
     def wavelen(self):
@@ -550,11 +550,11 @@ def z(self):
         return self.args[1]

     @property
-    def n(self):
+    def z_r(self):
         return self.args[2]

     @property
-    def z_r(self):
+    def n(self):
         return self.args[3]

     @property
```

Reply:
Thanks @smichr !